code/src/LeafletProvider/Features/Directions.ts

```diff
@@ -0,0 +1,417 @@
+                );
+                return new Promise((resolve) => resolve(0));
+            }
+            return new Promise((resolve) => {
+                // If there currentDistance is not NaN, this means it already has a value and the directions didn't change,
+                // so we can resolve the Promise with the same distance we had before
+                if (isNaN(this._currentDistance)) {
+                    this._bindDistance = this._routesFoundHandler.bind(
+                        this,
+                        resolve,
+                        ResolveType.GetDistance
+                    );
+                    this._directionsRenderer.on(
+                        'routesfound',
+                        this._bindDistance
+                    );
+                } else {
+                    resolve(this._currentDistance);
+                }
+            });
+        }
+
+        public getTotalDurationFromDirection(): Promise<number> {
+            // Validate if the directionsRenderer exists
+            if (!this._hasDirectionsRenderer()) {
+                OSFramework.Helper.ThrowError(
+                    this._map,
+                    OSFramework.Enum.ErrorCodes.API_FailedNoPluginDirections
+                );
+                return new Promise((resolve) => resolve(0));
+            }
+            return new Promise((resolve) => {
+                // If there currentDuration is not NaN, this means it already has a value and the directions didn't change,
+                // so we can resolve the Promise with the same duration we had before
+                if (isNaN(this._currentDuration)) {
+                    this._bindDuration = this._routesFoundHandler.bind(
+                        this,
+                        resolve,
+                        ResolveType.GetDuration
+                    );
+                    this._directionsRenderer.on(
+                        'routesfound',
+                        this._bindDuration
+                    );
+                } else {
+                    resolve(this._currentDuration);
+                }
+            });
+        }
+
+        public removeRoute(): OSFramework.OSStructures.ReturnMessage {
+            this.setState(false);
+            const returnigMessage =
+                new OSFramework.OSStructures.ReturnMessage();
+            if (!this._hasDirectionsRenderer()) {
+                returnigMessage.isSuccess = true;
+            } else {
+                returnigMessage.code =
+                    OSFramework.Enum.ErrorCodes.API_FailedRemoveDirections;
+            }
+            return returnigMessage;
+        }
+
+        public setPlugin(
+            provider: Constants.Directions.Provider | string,
+            apiKey: string
+        ): OSFramework.OSStructures.ReturnMessage {
+            if (this._directionsRenderer !== undefined) {
+                this.removeRoute();
+            }
```

Comment:
não temos de fazer unload do plugin anyhow? just confirming!

Reply:
What would be the point of unloading it?

Reply:
what happens when we start loading multiple plugins simultaneously? Sounds strange. Unloading would be to call it's destroy or so?

Reply:
If multiple loadPlugin actions get called simultaneously, that wouldn't be the best approach to do directions right? 
In my opinion if the developer wants to loadPlugin multiple times, that would mean that he must not want to use the previously loaded one anymore... Also, if the developers wants to set 2 plugins and then use 2 setDirections with one for the 1st plugin and the other for the 2nd plugin... That wouldn't be possible without having a new parameter for the setDirections...